modules/tinymce/src/core/main/ts/fmt/MatchFormat.ts

```diff
@@ -67,65 +67,65 @@ const matchName = (dom: DOMUtils, node: Node, format) => {
   }
 };
 
-const matchItems = (dom: DOMUtils, node: Node, format, itemName: string, similar: boolean, vars: FormatVars) => {
-  let key, value;
+const matchItems = (dom: DOMUtils, node: Node, format: Format, itemName: string, similar: boolean, vars: FormatVars): boolean => {
   const items = format[itemName];
-  let i;
 
   // Custom match
   if (format.onmatch) {
-    return format.onmatch(node, format, itemName);
+    const onMatch: (node: Node, format: Format, name: string) => boolean = format.onmatch;
+    return onMatch(node, format, itemName);
   }
 
   // Check all items
   if (items) {
     // Non indexed object
     if (typeof items.length === 'undefined') {
-      for (key in items) {
+      for (const key in items) {
         if (Obj.has(items, key)) {
-          if (itemName === 'attributes') {
-            value = dom.getAttrib(node, key);
-          } else {
-            value = FormatUtils.getStyle(dom, node, key);
+          const value = itemName === 'attributes' ? dom.getAttrib(node, key) : FormatUtils.getStyle(dom, node, key);
+          const expectedValue = FormatUtils.replaceVars(items[key], vars);
+
+          if (!value && !expectedValue) {
```

Comment:
As this is new code, can we please not use falsy checks here and instead use the relevant `Type` functions. If you're keen we can probably change the other could below as well I guess?

- [ ] Avoid falsy checks please

Reply:
Done — I'm slightly worried there will be something here that depended on empty strings being falsey, but that's what tests are for I guess.